authui/src/authflowv2/botprotection/cloudflareTurnstile.ts

```diff
@@ -34,19 +34,33 @@ export class CloudflareTurnstileController extends Controller {
   declare siteKeyValue: string;
   declare langValue: string;
   declare widgetTarget: HTMLDivElement;
+  declare widgetContainer: HTMLDivElement | undefined;
 
+  hasExistingWidget = () => {
+    return this.widgetContainer != null;
+  };
   resetWidget = () => {
-    window.turnstile.reset(this.widgetTarget);
+    if (!this.hasExistingWidget()) {
+      return;
+    }
+    window.turnstile.reset(this.widgetContainer);
   };
 
   connect() {
     window.turnstile.ready(() => {
-      // Do not render widget if any existing children
-      if (this.widgetTarget.childElementCount !== 0) {
+      if (this.hasExistingWidget()) {
         return;
       }
       const colorScheme = getColorScheme();
-      window.turnstile.render(this.widgetTarget, {
+
+      // Note how we wrap an extra layer of div here, because on cleanup we can just remove this extra layer of div.
+      // container-container
+      //   container <-- can just remove this on cleanup
+      //     widget
+      const widgetContainer = document.createElement("div");
+      this.widgetContainer = widgetContainer;
+      this.widgetTarget.appendChild(widgetContainer);
+      window.turnstile.render(widgetContainer, {
```

Comment:
Store widgetID in this controller. According to the documentation, the API works on ID, not HTMLElement. https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#remove-a-widget